relay-sampling/src/lib.rs

```diff
@@ -840,60 +786,40 @@ impl DynamicSamplingContext {
 
         adjusted_sample_rate
     }
+}
 
-    /// Returns whether a trace should be retained based on sampling rules.
-    ///
-    /// If [`SamplingResult::NoDecision`] is returned, then no rule matched this trace. In this
-    /// case, the caller may decide whether to keep the trace or not. The same is returned if the
-    /// configuration is invalid.
-    pub fn should_keep(&self, ip_addr: Option<IpAddr>, config: &SamplingConfig) -> SamplingResult {
-        if let Some(rule) = get_matching_trace_rule(config, self, ip_addr, RuleType::Trace) {
-            let adjusted_sample_rate = self.adjusted_sample_rate(rule.sample_rate);
-            let rate = pseudo_random_from_uuid(self.trace_id);
-
-            if rate < adjusted_sample_rate {
-                SamplingResult::Keep
-            } else {
-                SamplingResult::Drop(rule.id)
-            }
-        } else {
-            SamplingResult::NoDecision
+pub fn get_matching_trace_rule<'a>(
```

Comment:
Please add a descriptive doc comment to the public interface of the sampling crate, particularly mentioning what "matching" means here.

Aside, could we make this an associated function of either `SamplingConfig` or `DynamicSamplingContext`? Based on this interface, it looks like this is best placed on `&SamplingConfig`.